### PR TITLE
New Blog Post - Help CSS nesting pick syntax

### LINKED ISSUE #3223: [CSS] Help CSS nesting pick a syntax
* Proposed title: Help CSS nesting pick a syntax
* Proposed by: argyle
* Related areas: CSS
* Content type: Case Study (https://web.dev/handbook/case-study-guidelines/)
* [Draft](https://docs.google.com/document/d/1gRFk-_ctRaq58iGEOAr2dvPWz1P2IcT_XIbL2Hno3yc/edit?resourcekey=0-pM_qjayXlhHDrgy9AbFDqA#heading=h.cooyn6pp6jtu)
* Content summary: Two competing syntaxes need your help in determining which should be championed through to a spec candidate.

### PR DESCRIPTION
**Pull request by argyleink**
Fixes #3223

only weird thing in the post is having nunjucks ignore the bracketed syntax, since it thinks it's variables it needs to consume. i also added a small style tag to assist in titling code snippet sections, lemme know if that's not cool.

**todo**:

- [x] make a survey to collect user responses
- [x] add link to survey at the bottom of the post
- [x] consider alternative syntax comparison presentation
- [x] @mirisuzanne to review author data
- [x] consider moving "Equivalent CSS" to the beginning of the section

### CODE REVIEW

File: site/en/blog/help-css-nesting/index.md [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+  color: deeppink;
+}
+```
+
+An official CSS version of this syntax is being strongly considered and we have
+a split in preference that we'd like to employ the help of the community to
+break the tie. The rest of this post will be introducing the syntax options so
+you can feel informed to take a survey at the end.
+
+## Overview of what we're comparing
```
Comment (mirisuzanne) [inline site/en/blog/help-css-nesting/index.md:62]: Would it help to explain why the syntax above (used by pre-processors) isn't an option?

File: site/en/blog/help-css-nesting/index.md [hunk 2]
```diff
@@ -0,0 +1,776 @@
...
+layout: layouts/blog-post.njk
+title: Help CSS nesting pick syntax
+description: >
+  Two competing syntaxes need your help in determining which should be championed through to a spec candidate.
+subhead: >
+  Two competing syntaxes need your help in determining which should be championed through to a spec candidate.
+date: 2022-07-14
+authors:
+  - argyle
+  - miriamsuzanne
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/help-css-nesting/index.md:11]: you're totally a part of it imo 🙂 plenty of influence, enough to be a co-author.

File: site/en/blog/help-css-nesting/index.md (hunk 1 above)
Comment (argyleink) [reply site/en/blog/help-css-nesting/index.md:62]: sounds reasonable to me. do you happen to have a link we can provide that goes over it? or maybe want to offer a blurb?

Comment (mirisuzanne) [reply site/en/blog/help-css-nesting/index.md:62]: I documented some of it here: https://github.com/sass/sass/issues/3030

Not sure there's any reason to link there, but you could borrow some of the language.

Comment (argyleink) [reply site/en/blog/help-css-nesting/index.md:96]: added a section about it. thoughts? it's a tough one to touch on, hope i did ok

**Comment by argyleink** [comment]
ready for review 👍🏻

**Comment by rachelandrew** [comment]
I think @mirisuzanne needs to sign the CLA, the bot is unhappy.

File: site/en/blog/help-css-nesting/index.md [hunk 3]
```diff
@@ -0,0 +1,810 @@
+---
+layout: layouts/blog-post.njk
+title: Help CSS nesting pick syntax
+description: >
+  Two competing syntaxes need your help in determining which should be championed through to a spec candidate.
+subhead: >
+  Two competing syntaxes need your help in determining which should be championed through to a spec candidate.
+date: 2022-07-14
```
Comment (rachelandrew) [inline site/en/blog/help-css-nesting/index.md:8]: update the date before merging :)

File: site/en/blog/help-css-nesting/index.md [hunk 4]
```diff
@@ -0,0 +1,810 @@
+---
+layout: layouts/blog-post.njk
+title: Help CSS nesting pick syntax
+description: >
+  Two competing syntaxes need your help in determining which should be championed through to a spec candidate.
```
Comment (rachelandrew) [inline site/en/blog/help-css-nesting/index.md:5]:
```suggestion
  Two competing syntaxes need your help in determining which should be championed through to a specification candidate.
```

File: site/en/blog/help-css-nesting/index.md [hunk 5]
```diff
@@ -0,0 +1,810 @@
...
+subhead: >
+  Two competing syntaxes need your help in determining which should be championed through to a spec candidate.
```
Comment (rachelandrew) [inline site/en/blog/help-css-nesting/index.md:7]:
```suggestion
  Two competing syntaxes need your help in determining which should be championed through to a specification candidate.
```

File: site/en/blog/help-css-nesting/index.md [hunk 6]
```diff
@@ -0,0 +1,810 @@
+---
+layout: layouts/blog-post.njk
+title: Help CSS nesting pick syntax
```
Comment (rachelandrew) [inline site/en/blog/help-css-nesting/index.md:3]:
```suggestion
title: Help pick a syntax for CSS nesting
```

Comment (rachelandrew) [reply site/en/blog/help-css-nesting/index.md:3]: I think this reads better.

**Review by rachelandrew — APPROVED** [review]
There's a few tiny nits, but otherwise this LGTM. Just needs the CLA sorting.

Let me know if you want me to queue up a tweet from ChromiumDev.

File: site/en/blog/help-css-nesting/index.md (hunk 3 above)
Comment (argyleink) [reply site/en/blog/help-css-nesting/index.md]: i set it for 7-22 (tomorrow) but we can adjust as needed
thanks!

**Comment by korenevskiy** [comment]
Hey
Why would NESting use the nested type?
You can do the same syntacticly without nesting, but have a nesting.

```CSS
div.alfa{
	color: blue;
}
:& .title{
	color: red;
}
:& .info{
	color: green;
}
:&(1) .content{
	color: gray;
}
:&(1) .content:media(width < 600px){
	color: gray;
}
```

Equivalent CSS
```CSS
div.alfa{
	color: blue;
}
div.alfa .title{
	color: red;
}
div.alfa .info{
	color: green;
}
div.alfa  .info  .content{
	color: gray;
}
@media(width < 600px){
div.alfa  .info  .content{
	color: gray;
}
}
```
As you can see, syntactic compatibility is observed.
But at the same time it has all the advantages of a new way.

It is also convenient to make an attachment of any depth without changing the parser.

**Pull request merged**